### PR TITLE
feat: more apis for dataset generation

### PR DESCRIPTION
**Pull request by StochasticRomanAgeev**
Adding more apis for dataset generation

### CODE REVIEW

File: examples/dataset_generation/create_alpaca_dataset.ipynb [hunk 1]
```diff
@@ -8,106 +8,97 @@
     "# Generate dataset for ALPACA\n",
     "\n",
     "## Use token from OpenAI to generate dataset for finetuning\n",
-    "### You can get token from [here](https://beta.openai.com/docs/api-reference/authentication)"
+    "### You can get token from [here](https://beta.openai.com/docs/api-reference/authentication)\n",
```
Comment (sarthaklangde) [inline examples/dataset_generation/create_alpaca_dataset.ipynb:11]: @StochasticRomanAgeev I want to see example in the same notebook of how to use AI 21 and Cohere as well. It is not clear. You can comment out the code if you want and only keep OpenAI by default but the examples need to show that

File: examples/dataset_generation/create_alpaca_dataset.ipynb [hunk 2]
```diff
@@ -8,106 +8,97 @@
...
+    "pycharm": {
+     "is_executing": true
     }
-   ],
+   },
+   "outputs": [],
    "source": [
     "from xturing.datasets import InstructionDataset\n",
-    "from xturing.model_apis import openai"
+    "from xturing.model_apis import Davinci"
```
Comment (sarthaklangde) [inline examples/dataset_generation/create_alpaca_dataset.ipynb:26]: I don't like this. What if Cohere also has a model called Davinci? We should reflect the change properly. I would prefer something like:

```from xturing.model_apis.openai import Davinci```

Comment (sarthaklangde) [reply examples/dataset_generation/create_alpaca_dataset.ipynb:26]: Wait why is this different from `TextGenerationAPI` mentioned below?

Comment (sarthaklangde) [reply examples/dataset_generation/create_alpaca_dataset.ipynb:26]: Ok, it seems you just need to fix this, right?

**Review by sarthaklangde — CHANGES_REQUESTED** [review]
Additional todo:
- Change README.md roadmap to show that dataset generation can be done with OpenAI, Cohere and AI ...

File: examples/dataset_generation/create_alpaca_dataset.ipynb (hunk 2 above)
Comment (StochasticRomanAgeev) [reply examples/dataset_generation/create_alpaca_dataset.ipynb:26]: TextGenerationAPI is base class, and other api classes are using it

File: README.md [hunk 3]
```diff
@@ -107,6 +107,7 @@ Please submit your performance results on other GPUs.
 - [x] Dataset generation using self-instruction
 - [x] 2x more memory-efficient fine-tuning vs LoRA and unsupervised fine-tuning
 - [x] INT8 low-precision fine-tuning support
+- [x] Cohere and AI21 APIs support for dataset generation
```
Comment (sarthaklangde) [inline README.md:110]:
```suggestion
- [x] Supports OpenAI, Cohere and AI21 Studio model APIs for dataset generation
```

**Review by sarthaklangde — CHANGES_REQUESTED** [review]

**Pull request merged**